src/pipelinedefinition.hpp

```diff
@@ -71,13 +73,37 @@ struct NodeInfo {
 };
 
 class PipelineDefinition {
+    struct ValidationResultNotifier {
+        ValidationResultNotifier(PipelineDefinitionStatus& status) :
+            status(status) {}
+        ~ValidationResultNotifier() {
+            if (passed) {
+                status.handle(ValidationPassedEvent());
+            } else {
+                status.handle(ValidationFailedEvent());
+            }
+        }
+        bool passed = false;
+
+    private:
+        PipelineDefinitionStatus& status;
+    };
+
     const std::string pipelineName;
     std::vector<NodeInfo> nodeInfos;
     pipeline_connections_t connections;
-    std::set<std::pair<const std::string, model_version_t>> subscriptions;
+
     std::atomic<uint64_t> requestsHandlesCounter = 0;
     std::shared_mutex loadMtx;
 
+    std::condition_variable loadedNotify;
+
+protected:
+    PipelineDefinitionStatus status;
+
+private:
+    std::set<std::pair<const std::string, model_version_t>> subscriptions;
```

Comment:
Why have you moved this field from one private sector to another?

Reply:
I wanted to divide the variables by their usage purpose in case further refactor splits this class.